controllers/machinehealthcheck_controller.go

```diff
@@ -251,9 +251,13 @@ func (r *MachineHealthCheckReconciler) reconcile(ctx context.Context, logger log
 	errList := []error{}
 	for _, t := range unhealthy {
 		condition := conditions.Get(t.Machine, clusterv1.MachineHealthCheckSuccededCondition)
-		logger.Info("Target has failed health check, marking for remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
 
-		conditions.MarkFalse(t.Machine, clusterv1.MachineOwnerRemediatedCondition, clusterv1.WaitingForRemediation, clusterv1.ConditionSeverityWarning, "MachineHealthCheck failed")
+		if annotations.IsPaused(cluster, t.Machine) {
+			logger.Info("Machine has failed health check, but machine is paused so skipping remediation", "target", t.string(), "reason", condition.Reason, "message", condition.Message)
```

Comment:
My only concern is that the user could be confused by a machine being unhealthy and remediation requests not being generated (with the reason explained in logs only). 
Might be we should consider adding a note in the in unhealthy condition message, wdyt?

Reply:
Hmm, do you think that a user wouldn't know that they paused their machine?

Reply:
Maybe paused should be a condition too?

Reply:
Let's open a different issue to add a condition for paused in v0.4, for now this seems fine

Reply:
I was more thinking to a whole cluster paused, but I'm ok in tackling UX concern in a separated issue

Reply:
If the whole cluster is paused it will not reach this code because it is being handled [here](https://github.com/kubernetes-sigs/cluster-api/blob/3900bb1cb2cb77c6ceebeb747647b22db2817dba/controllers/machinehealthcheck_controller.go#L139-L143)

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/3450